src/commands/lambda/instrument.ts

```diff
@@ -177,15 +182,46 @@ export class InstrumentCommand extends Command {
     const tracingEnabled = this.convertStringBooleanToBoolean(true, this.tracing, this.config.tracing)
     const logLevel = this.logLevel ?? this.config.logLevel
 
+    const service = this.service ?? this.config.service
+    const environment = this.environment ?? this.config.environment
+    const version = this.version ?? this.config.version
+
+    const tagsMap: {[key: string]: string | undefined} = {
+      environment,
+      service,
+      version,
+    }
+    let tagMissing = false
+    for (const [tag, value] of Object.entries(tagsMap)) {
+      if (!value) {
+        tagMissing = true
+        this.context.stdout.write(`No value found for the ${tag} tag.\n`)
+      }
+    }
+    if (tagMissing) {
+      this.context.stdout.write('It is recommended to set tags, try to do it later.\n')
+    }
+
+    const extraTags = this.extraTags ?? this.config.extraTags
+    if (extraTags && !extraTags.match(EXTRA_TAGS_REG_EXP)) {
```

Comment:
Could be great to extract this part to a pure function with a string + regex as input and a boolean as a result.
That way you could test some valid/invalid inputs against the resulting boolean

Reply:
Also we are lowercasing the tags downstream in layers/extensions but we should do it here as well